app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,81 @@
+class PicturesController < ApplicationController
+  before_action :set_picture, only: [:show, :edit, :update, :destroy]
+  before_filter :authenticate_user!, except: [:index, :show]
+  before_action :correct_user, only: [:edit, :update, :destroy]
+
+  # GET /pictures
+  # GET /pictures.json
+  def index
+    @pictures = Picture.all
+  end
+
+  # GET /pictures/1
+  # GET /pictures/1.json
+  def show
+  end
+
+  # GET /pictures/new
+  def new
+    @picture = current_user.pictures.build
+  end
+
+  # GET /pictures/1/edit
+  def edit
+  end
+
+  # POST /pictures
+  # POST /pictures.json
+  def create
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for create is too high. [15.07/15] (http://c2.com/cgi/wiki?AbcMetric)